featurematrix.go

```diff
@@ -442,6 +442,23 @@ func (fm *FeatureMatrix) LoadCases(data *csv.Reader, rowlabels bool) {
 
 }
 
+// Makes a copy of the feature matrix
+func (fm *FeatureMatrix) Copy() *FeatureMatrix {
+	newmap := make(map[string]int)
+	newdata := make([]Feature, 0, len(fm.Data))
+	for _, feature := range fm.Data {
+		newdata = append(newdata, feature.Copy())
```

Comment:
can we do `newdata[i] = feature.Copy()` to avoid `append`?

Reply:
sure